coredb-operator/src/controller.rs

```diff
@@ -139,6 +147,50 @@ impl CoreDB {
             .map_err(Error::KubeError)?;
         Ok(Action::await_change())
     }
+
+    async fn primary_pod(&self, client: Client) -> Result<Pod> {
```

Comment:
When I'm returning Result, I ought to also handle errors. And when I'm unwrapping, I should pass errors up with '?'